### DISCUSSION TITLE
Use local history for expanding the Speed Dial

### DISCUSSION THREAD

**Discussion by Albert221** [darioielardi/flutter_speed_dial#100] [Q&A] [answered]
Hi!

- Right now if you expand the Speed Dial children and want to go back using the device Back button, you go back to the previous screen, not close the Speed Dial.
- The same behavior happens with `Navigator.pop()`. When I'm deep in my navigation stack and I want to go back to the route with the Speed Dial, I can either choose to pop to the Speed Dial opened route or to the previous screen. I can't pop to the screen with the Speed Dial but with it being closed.

There is - although - one thing in Flutter that isn't obvious and will help you in fixing that issue/adding better navigation support that you can use and that's local history.

[addLocalHistoryEntry](https://api.flutter.dev/flutter/widgets/LocalHistoryRoute/addLocalHistoryEntry.html)

Please, learn about it and use it in your library. Apart from that, it's really great.

If you can't work on that right now let me know. I may create and submit a PR myself.

Cheers!

**Comment by prateekmedia** [comment, marked as answer]
You can checkout below guide
https://github.com/darioielardi/flutter_speed_dial#close-on-willpop

It tells you how to integrate closing of speeddial before closing the screen.
